common-pages/src/TimeFilter.tsx

```diff
@@ -0,0 +1,715 @@
+                centerTime: t.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                startTime: start.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                endTime: end.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                timeWindowUnits: 6,
+                windowSize: 12,
+                halfWindowSize: 6,
+            }
+        }
+    },
+    {
+        label: 'Last Year', createFilter: (tz) => {
+            const offset = momentTZ.tz(moment.utc().startOf('year').subtract(1, 'year').format('YYYY-MM-DDTHH:mm:ss.SSSSS'), tz).utcOffset();
+            const t = moment.utc().add(offset, 'minute').startOf('year').subtract(0.5, 'year');
+            
+            const [start, end] =  getStartEndTime(t, 6, 'M');
+            return {         
+                centerTime: t.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                startTime: start.add(1, 'day').format('MM/DD/YYYY HH:mm:ss.SSS'),
+                endTime: end.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                timeWindowUnits: 6,
+                windowSize: 12,
+                halfWindowSize: 6,
+            }
+        }
+    },
+    {
+        label: 'Last 365 Days', createFilter: (tz) => {
+            const offset = momentTZ.tz(moment.utc().startOf('day').subtract(182.5, 'days').format('YYYY-MM-DDTHH:mm:ss.SSSSS'), tz).utcOffset();
+            const t = moment.utc().add(offset, 'minute').startOf('day');
+            t.subtract(182.5, 'days');
+            const [start, end] =  getStartEndTime(t, 182.5, 'd');
+            return {         
+                centerTime: t.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                startTime: start.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                endTime: end.format('MM/DD/YYYY HH:mm:ss.SSS'),
+                timeWindowUnits: 4,
+                windowSize: 365,
+                halfWindowSize: 182.5,
+            }
+        }
+    }
+];
+
+
+
+
+
+function Row(props: React.PropsWithChildren<{isHorizontal: boolean}>){
+    if (props.isHorizontal){
+        return <div className='row'>{props.children}</div>
+    }
+    return <>{props.children}</>
+}
+
+
+
+
+
+const TimeFilter = (props: IProps) => {
+    const [activeQP, setActiveQP] = React.useState<number>(-1);
+    const [filter, setFilter] = React.useState<ITimeFilter>({
+        centerTime: props.filter.centerTime,
+        startTime: props.filter.startTime,
+        endTime: props.filter.endTime,
+        timeWindowUnits: props.filter.timeWindowUnits,
+        windowSize: props.filter.windowSize,
+        halfWindowSize: props.filter.halfWindowSize,
+    });
+
+
+    React.useEffect(() => {
+        if (isEqual(filter, props.filter))
+            return;
+        props.setFilter({
+            centerTime: filter.centerTime,
+            startTime: filter.startTime,
+            endTime: filter.endTime,
+            windowSize: filter.windowSize,
+            timeWindowUnits: filter.timeWindowUnits,
+            halfWindowSize: filter.halfWindowSize
+        });
+    }, [filter])
```

Comment:
I am not convinced this is neccesarry but if you cnahe typing of `ITimeFilter` it may be neccesarry to change to an object that includes all options here. - would recommend makng these sepperate states (start/end), (center/duration) etc